include/CMakeLists.txt

```diff
@@ -6,13 +6,11 @@
 #
 add_library(simdjson-headers INTERFACE)
 target_compile_features(simdjson-headers INTERFACE cxx_std_11) # headers require at least C++11
-target_include_directories(simdjson-headers INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+target_include_directories(simdjson-headers INTERFACE
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCDIR}>)
 
-install(TARGETS simdjson-headers
-        EXPORT simdjson-headers-config
-        INCLUDES DESTINATION include)
-install(EXPORT simdjson-headers-config
-        FILE simdjson-headers-config.cmake
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/simdjson)
+install(TARGETS simdjson-headers EXPORT simdjson-config INCLUDES DESTINATION include)
 install(DIRECTORY simdjson DESTINATION include FILES_MATCHING PATTERN *.h)
+install(DIRECTORY simdjson DESTINATION include FILES_MATCHING PATTERN *.hpp)
```

Comment:
NOT something to change in this PR, but:

I feel like we shouldn't be installing the include files at all. It's legacy, and @lemire can comment more, but I don't think they are public interface. The amalgamated header is the only one we should publish (you can see where that is generated in singleheader/CMakeLists.txt).

However, I still want to link the tools and tests to the include/ and src/ files, not the amalgamated versions, because I want file and line numbers for debugging :) I am fairly sure you know more cmake than I do (my experience level being about 3-4 weeks at this point), so I'm hoping you can advise whether this dream is of the pipe variety :)

Reply:
Just to be clear, you are saying that we should use include/ and src/ folders for internal use (e.g. tests and tools) and amalgamated header file for installation. If so, I think I can do that.